### PR TITLE
Add toot export

### PR DESCRIPTION
**Pull request by ddevault**
Import can come separately. Requires some discussion and buy-in from the maintainers on a strategy.

### CODE REVIEW

**Comment by Gargron** [comment]
Commendable effort, but my concern is: how does this behave when someone has, say, 90,000 toots?

The file returned would be huge, but most importantly, the request to generate it would take so long it would time out. Can you think of any ways to mitigate this?

**Comment by SirCmpwn** [comment]
Perhaps we can set an upper limit on it, like it only exports up to the most recent 5000 toots?

**Comment by papjul** [comment]
Maybe we can get inspired by how Twitter handles it?
It generates in a background task and when it’s ready you receive an email to download zip file. You should also put a limit of a generation every X hours so the same file is served if requested multiple times during a certain period of time.

**Review by mjankowski — CHANGES_REQUESTED** [review]
Needs rebase.

File: app/controllers/settings/exports_controller.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 require 'csv'
+include RoutingHelper
```
Comment (mjankowski) [inline app/controllers/settings/exports_controller.rb]: What is this adding?

**Comment by ashfurrow** [comment]
It sounds like there are some technical details that need to be resolved before we can review this. I'm going to suggest we move this into a issue instead of a pull request, and we can resolve things there?

**Comment by Gargron** [comment]
Closing this down, toot export is on the backburner, this is incomplete, and has been inactive for a while. But uhh, in case you are reading this, OP, I do thank you for the effort!

**Pull request closed without merging**